Components/AllEventPage/EventBox.tsx

```diff
@@ -1,29 +1,89 @@
+import { Alert, AlertTitle } from "@mui/material";
+import { useEffect, useState } from "react";
 import classes from "./EventBox.module.css";
 import EventBoxContent from "./EventBoxContent";
 import EventElement from "./EventElement";
 
+
+interface Organizer {
+  shortName: string;
+  displayName: string;
+}
+
+
+interface Registration {
+  link: string;
+  type: string;
+}
+
+interface Event {
+  id: string;
+  subtitle: string;
+  displayName: string;
+  description: string;
+  endAt: number;
+  startAt: number;
+  organizers: Organizer[];
+  registration: Registration;
+}
+
+interface EventResponse {
+  isError: boolean;
+  data: Event[];
+}
+
+const ErrorAlert = (props: { errorMessage: string }) => {
+  return (
+    <Alert severity="error">
+      <AlertTitle>Error while fetching events!</AlertTitle>
+      {props.errorMessage}
+    </Alert>
+  );
+};
+
+const epochToString = (epoch: number) => new Date(epoch).toLocaleString("en-IN");
+
 function EventBox() {
+  const [events, setEvents] = useState<Event[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>("");
+
+  useEffect(() => {
+    fetch('https://asia-south1-nsccpccoe.cloudfunctions.net/events')
+      .then(res => {
+        if (!res.ok) {
+          throw res.statusText;
+        }
+        return res.json();
+      })
+      .then((eventResponse: EventResponse) => {
+        if (eventResponse.isError) {
+          setErrorMessage("Unknown error occured while fetching event details");
+        } else {
+          setEvents(eventResponse.data);
+        }
+      })
+      .catch(error => {
+        setErrorMessage(error)
+      });
+  }, []);
+
+
   return (
     <div className={classes.container}>
       <div className={classes.content}>
         <EventBoxContent />
       </div>
       <div className={classes.events}>
-        <EventElement
-          type="Web Development + Marketing Hackathon"
-          heading="WebXplore"
-          duration="22/01/2023 8:00 PM - 29/01/2023 11:59 PM IST"
-        />
-        <EventElement
-          type="Competitive Programming Contest"
-          heading="CodeHive"
-          duration="26/01/2023, 02:00 PM - 05:00 PM IST"
-        />
-        <EventElement
-          type="Competitive Programming + Hiring Contest"
-          heading="CodeRush-X"
-          duration="29/01/2023, 09:00 PM - 12:00 AM IST"
-        />
+        {
+          (errorMessage.length != 0) ? <ErrorAlert errorMessage={errorMessage} /> : events.map(event => {
```

Comment:
Instead of checking `errorMessage.length` I guess it would be better if you check `isError`

Reply:
`isError` wont available in if request fails. for example if you change endpoint from /events to /evens you get `403 forbidden` with no body. this is handled with 
```ts
  if (!res.ok) {
          throw res.statusText;
        }
        return res.json();
.
.
.catch(error => {
        setErrorMessage(error)
      }); 
```

Reply:
I was using on more state variable `isError` but that feels redundant. with that approach I need to set `isError` state everytime I change `errorMessage`

Reply:
okay! thanks for clarifying!!! Nice approach 👍🏻